LibreNMS/OS/Procurve.php

```diff
@@ -48,4 +52,104 @@ public function pollOS(DataStorageInterface $datastore): void
             $this->enableGraph('fdb_count');
         }
     }
+
+    private static function pivotTable(array $table): array
+    {
+        $res = [];
+        foreach ($table as $column => $data) {
+            foreach ($data as $index => $value) {
+                $res[$index][$column] = $value;
+            }
+        }
+
+        return $res;
+    }
+
+    public function pollNac()
+    {
+        $nac = new Collection();
+
+        $enabled = SnmpQuery::mibs(['IEEE8021-PAE-MIB'])->hideMib()->enumStrings()->get('dot1xPaeSystemAuthControl.0')->value();
+        if ($enabled !== 'enabled') {
+            return $nac;
+        }
+
+        $rowSet = [];
+        $ifIndex_map = $this->getDevice()->ports()->pluck('port_id', 'ifIndex');
+
+        $table = SnmpQuery::mibDir('hp')->mibs(['HP-DOT1X-EXTENSIONS-MIB'])->hideMib()->enumStrings()->walk('hpicfDot1xSMAuthConfigTable')->table();
+        $table = self::pivotTable($table);
```

Comment:
Have you tried to use `->table(2) ` instead ? That would probably give you the correct table and allow you to remove pivotTable function.

Reply:
Good catch, I hadn't tried that.

Refactored as suggested.